### ISSUE TITLE
Issue with JUnit 5 M3 console launcher with JAR files

### ISSUE THREAD

**Issue by danilofaria** [junit-team/junit5#613]
I would like to run the console launcher using JAR files instead of .class files for the tests. According to the M3 release notes, it should work ("Classpath scanning now supports JAR files.")
It won't pick up the tests when I run the command, though. Do I need to add any other flags or do something else?

**Comment by nicolaiparlog** [comment]
I'm not sure whether this still applies but for a while you had to add an engine to the console launcher's class path. Can you successfully run `.class` tests with the same configuration? Does the command line contain any complaints talking about engines?

**Comment by danilofaria** [comment]
@nicolaiparlog The command I'm using is 
`java -cp "lib/*" org.junit.platform.console.ConsoleLauncher -cp src --scan-class-path --hide-details --include-classname "^.*IT?$"`
It does work with `.class` tests, that's what we're using right now.
However, when I switch to JAR files, it does not pick up the tests.
Part of the output is:
```
INFO: Discovered TestEngines with IDs: [junit-jupiter]

Test run finished after 30 ms
[         1 containers found      ]
[         0 containers skipped    ]
[         1 containers started    ]
[         0 containers aborted    ]
[         1 containers successful ]
[         0 containers failed     ]
[         0 tests found           ]
[         0 tests skipped         ]
[         0 tests started         ]
[         0 tests aborted         ]
[         0 tests successful      ]
[         0 tests failed          ]
```

**Comment by marcphilipp** [comment]
We don't scan JARs by default. Please add the path of your JAR file as argument to `--scan-class-path`.

**Comment by marcphilipp** [comment]
This definitely needs better documentation… 🤔

**Comment by danilofaria** [comment]
Hello @marcphilipp, I tried adding the path as you said (` --scan-class-path "src/testing-app-tests.jar" `), but it still did not pick it up.

**Comment by danilofaria** [comment]
Hello @marcphilipp @sbrannen @nicolaiparlog, can you confirm whether what I'm trying to do is possible? I've already tried what was suggested here but still does not work. Your help is immensely appreciated.

**Comment by marcphilipp** [comment]
I am currently on vacation but will check when I return tomorrow.

**Comment by marcphilipp** [comment]
I'm afraid "Classpath scanning now supports JAR files" is misleading and my previous comment was wrong. At the moment we only support resolving package selectors, not JARs as classpath roots (see #399).

So, you have to switch from `--scan-class-path` to `--select-package $YOUR_ROOT_PACKAGE`.

However, I think explicitly selecting a JAR to be scanned like I suggested above makes sense and we should add this.

@junit-team/junit-lambda What do you guys think?

**Comment by danilofaria** [comment]
@marcphilipp So you're saying that if I switch from `--scan-class-path` to `--select-package $YOUR_ROOT_PACKAGE`, then the console launcher should work with my JAR file? 
I did that and it did not work. 
Or are you saying that JAR files are not support at all for the console launcher?

**Comment by marcphilipp** [comment]
Selecting a package that resides within a package does work; I tried that yesterday. Can you please post your complete command?

**Comment by danilofaria** [comment]
@marcphilipp the complete command looks like the following
```
java -cp "lib/*" org.junit.platform.console.ConsoleLauncher -cp src  --hide-details --select-package "com.companyname.bar.foo.testing"  --include-classname "^.*IT?$" 
```

**Comment by marcphilipp** [comment]
Is your test JAR still in src? Where do you add it to the classpath? `-cp src` only adds the src directory to the classpath.

**Comment by danilofaria** [comment]
@marcphilipp yes, the JAR is inside of the src directory. Do I need to provide any extra information?

**Comment by sbrannen** [comment]
@danilofaria, your JAR has to be on the classpath, not in a folder that happens to be on the classpath.

**Comment by danilofaria** [comment]
@marcphilipp @sbrannen I got it to work with the following command:
```
java -cp "lib/*" org.junit.platform.console.ConsoleLauncher -cp "src/jar_name.jar"  --hide-details --select-package "com.companyname.bar.foo.testing"  --include-classname "^.*IT?$" 
```
Thanks very much for the help.

**Comment by sbrannen** [comment]
@danilofaria, Great!

Thanks for letting us know you got it working.

**Comment by marcphilipp** [comment]
I've added a warning to the User Guide and clarified the release notes. Both changes have been published to http://junit.org/junit5/docs/5.0.0-M3/user-guide/ and http://junit.org/junit5/docs/current/user-guide/.

Adding the ability to scan JAR files as classpath roots will be addressed in #620.

**Comment by sbrannen** [comment]
@marcphilipp, wow... you even updated the _current_ online docs. Kudos!
